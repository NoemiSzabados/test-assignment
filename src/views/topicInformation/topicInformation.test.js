import renderer, {act} from "react-test-renderer";
import TopicInformation from "./index";
import topicsProvider from "../../data";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import ReactDOM from "react-dom";


Enzyme.configure({ adapter: new Adapter() });

describe('Topic Information', () => {
    it('snapshot test', () => {
        const component = renderer.create(
            <TopicInformation />,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

    });

    it('renders without crush', () => {
        const div = document.createElement("div");
        ReactDOM.render(<TopicInformation />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('topic information is hidden after init', () => {
        const wrapper = shallow(<TopicInformation />);
        expect(wrapper.find('.hidden').exists()).toBe(true);
    });

    it('topic information is hidden when topic is null', () => {
        const wrapper = shallow(<TopicInformation />);
        const topic = null;
        act(() => {
            topicsProvider.selectedTopicStream.next(topic);
        });
        expect(wrapper.find('.hidden').exists()).toBe(true);
    });

    it('topic information is visible when topic is not null', async () => {
        const component = renderer.create(
            <TopicInformation />,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
        const topic =
            {
                "id": "1751295897__Berlin",
                "label": "Berlin",
                "volume": 165,
                "type": "topic",
                "sentiment": {
                    "negative": 3,
                    "neutral": 133,
                    "positive": 29
                },
                "sentimentScore": 65,
                "burst": 13,
            };
        act(()=> {
            topicsProvider.selectedTopicStream.next(topic);
        })

        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
