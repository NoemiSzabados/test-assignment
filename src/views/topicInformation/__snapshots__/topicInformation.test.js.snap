// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topic Information snapshot test 1`] = `
<div
  className="infoWrapper hidden"
/>
`;

exports[`Topic Information topic information is visible when topic is not null 1`] = `
<div
  className="infoWrapper hidden"
/>
`;

exports[`Topic Information topic information is visible when topic is not null 2`] = `
<div
  className="infoWrapper"
>
  <div
    className="infoContainer"
  >
    <div>
      Information on topic "
      Berlin
      "
    </div>
    <div>
      Total mentions: 
      165
    </div>
    <div>
      <div>
        Positive mentions: 
        <span
          className="positiveMentions"
        >
          29
        </span>
      </div>
      <div>
        Neutral mentions: 
        133
      </div>
      <div>
        Negative mentions: 
        <span
          className="negativeMentions"
        >
          3
        </span>
      </div>
    </div>
  </div>
</div>
`;
